Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -1028,125 +1028,128 @@ extension MergedSQLiteBookmarks {
             return succeed()
         }
 
-        let deferred = Success()
-
-        if let err = self.local.db.transaction({ conn -> Bool in
-            // This is a little tortured because we want it all to happen in a single transaction.
-            // We walk through the accrued work items, applying them in the right order (e.g., structure
-            // then value), doing so with the ugly NSError-based transaction API.
-            // If at any point we fail, we abort, roll back the transaction (return false),
-            // and reject the deferred.
-
-            func change(_ sql: String, args: Args?=nil) throws {
-                if let err = conn.executeChange(sql, withArgs: args) {
-                    deferred.fillIfUnfilled(Maybe(failure: DatabaseError(err: err)))
-                    throw err
-                }
-            }
-
+        // This is a little tortured because we want it all to happen in a single transaction.
+        // We walk through the accrued work items, applying them in the right order (e.g., structure
+        // then value).If at any point we fail, we abort, roll back the transaction by throwing.
+        return local.db.transaction { conn -> Void in
             // So we can trample the DB in any order.
-            try change("PRAGMA defer_foreign_keys = ON")
-
+            if let err = conn.executeChange("PRAGMA defer_foreign_keys = ON") {
+                log.error("applyLocalOverrideCompletionOp(_:, itemSources:) encountered error: \(err.localizedDescription)")
+                throw err
+            }
+            
             log.debug("Deleting \(op.mirrorItemsToDelete.count) mirror items.")
-            var deleteError: NSError? = nil
-            op.mirrorItemsToDelete
-              .withSubsetsOfSize(BrowserDB.MaxVariableNumber) { guids in
-                guard deleteError == nil else {
-                    return
-                }
 
-                do {
+            try op.mirrorItemsToDelete
+                .withSubsetsOfSize(BrowserDB.MaxVariableNumber) { guids in
                     let args: Args = guids.map { $0 }
                     let varlist = BrowserDB.varlist(guids.count)
-
+                    
                     let sqlMirrorStructure = "DELETE FROM \(TableBookmarksMirrorStructure) WHERE parent IN \(varlist)"
-                    try change(sqlMirrorStructure, args: args)
-
+                    if let err = conn.executeChange(sqlMirrorStructure, withArgs: args) {
+                        log.error("applyLocalOverrideCompletionOp(_:, itemSources:) encountered error: \(err.localizedDescription)")
```

Comment:
I don't think we need to log.

You might also consider defining `conn.tryChange` that throws, simplifying this code.